### PR TITLE
add/drop track for user feature, working on ex handler

### PR DESCRIPTION
**Pull request by avvinidiktov**

### CODE REVIEW

File: src/main/java/com/gmail/avvinidiktov/capitalOne/Configuration/WebSecurityConfig.java [hunk 1]
```diff
@@ -27,16 +28,23 @@
...
-        http.csrf().disable()
+        http.csrf()
+                .disable()
                 .authorizeRequests()
                 .antMatchers("/api/auth/**")
                 .permitAll()
-                .antMatchers("/api/companies/")
-                .authenticated()
+                .antMatchers("/api/admin/**")
+                .hasRole(Role.USER.name())
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/Configuration/WebSecurityConfig.java:40]: user role for admin scope

File: src/main/java/com/gmail/avvinidiktov/capitalOne/DTO/AuthResponce.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 import lombok.NoArgsConstructor;
 
+import java.time.Instant;
+
+
 @Data
+@Builder
 @NoArgsConstructor
 @AllArgsConstructor
 public class AuthResponce {
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/DTO/AuthResponce.java:15]: Response

File: src/main/java/com/gmail/avvinidiktov/capitalOne/DTO/CompanyInfoDTO.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.gmail.avvinidiktov.capitalOne.DTO;
+
+public class CompanyInfoDTO {
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/DTO/CompanyInfoDTO.java:3]: Do we need an empty class

File: src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/CompanyTrack.java [hunk 4]
```diff
@@ -28,4 +28,15 @@
...
+
+    @Column(name="active")
+    private boolean active;
+
+    public void enable(){
+        this.active = true;
+    }
+
+    public void disable(){
+        this.active = false;
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/CompanyTrack.java:40]: do not good idea to use a logic in an Entity class. Entity has to be POJO

File: src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/User.java [hunk 5]
```diff
@@ -1,7 +1,7 @@
 package com.gmail.avvinidiktov.capitalOne.Entity;
 
 import com.gmail.avvinidiktov.capitalOne.Enumerable.Role;
-import lombok.Data;
+import lombok.*;
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/User.java:4]: please do not use imports with an asterisk. Could you please a number of imports in your IDE

File: src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/User.java [hunk 6]
```diff
@@ -11,8 +11,11 @@
 import java.util.List;
 import java.util.Set;
 
-@Data
+@Getter
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/User.java:14]: Why do remove '@Data'? You can use '@Data" with  '@AllArgsConstructor'

File: src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/User.java [hunk 7]
```diff
@@ -82,4 +86,16 @@ public boolean isEnabled() {
...
+        this.track.add(track);
+    }
+
+    public void dropTrack(CompanyTrack track){
+        this.track.remove(track);
+    }
+
+    public void addRole(Role role){
+        this.roles.add(role);
+    }
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/Entity/User.java:100]: the same additional logic

**Review by AlRezz — CHANGES_REQUESTED** [review]
Check comments

File: src/main/java/com/gmail/avvinidiktov/capitalOne/Configuration/WebSecurityConfig.java [hunk 8]
```diff
@@ -27,16 +28,23 @@
...
     private JwtAuthFilter jwtAuthFilter;
 
+    //TODO: fix roles issue
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http.csrf().disable()
+        http.csrf()
+                .disable()
                 .authorizeRequests()
                 .antMatchers("/api/auth/**")
```
Comment (AlRezz) [inline src/main/java/com/gmail/avvinidiktov/capitalOne/Configuration/WebSecurityConfig.java:37]: Could you please add a 'Home' page for any user

File: src/main/java/com/gmail/avvinidiktov/capitalOne/DTO/CompanyInfoDTO.java (hunk 3 above)
Comment (avvinidiktov) [reply src/main/java/com/gmail/avvinidiktov/capitalOne/DTO/CompanyInfoDTO.java:3]: forget to unchek file in commit menu

**Comment by avvinidiktov** [comment]
now for each lvl of sub we have static num of tracks

**Review by AlRezz — APPROVED** [review]

**Pull request merged**